### PR TITLE
feature: Implement Consistency decoder for SDv1VAE

### PR DESCRIPTION
**Pull request by rodSiry**
- [x] Check if the original implementation of consistency decoder decodes an image encoded from refiners SD1Autoencoder() 
- [x] Identify model architecture
- [x] Write chain module classes
- [x] Functional weight converter
- [x] Outer decoding loop & final test
- [ ] cleaning, refactoring, unit tests...

### CODE REVIEW

**Comment by rodSiry** [comment]
I think the feature is functional (results below)

Before further cleaning, I'd like to know : 
-Should I define a ConsistencyAutoencoder class (i.e. default encoder + consistency decoder) ?

Don't hesitate to provide additional feedback.

ground truth
Ground Truth
default decoder
Default decoder
consistency decoder (diffusers)
Consistency decoder (from diffusers)
consistency decoder (ours)
Consistency decoder (from refiners)

File: src/refiners/foundationals/latent_diffusion/solvers/consistency_decoder_solver.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        raise ValueError(f"Unsupported alpha_tranform_type: {alpha_transform_type}")
+
+    betas = []
+    for i in range(num_diffusion_timesteps):
+        t1 = i / num_diffusion_timesteps
+        t2 = (i + 1) / num_diffusion_timesteps
+        betas.append(min(1 - alpha_bar_fn(t2) / alpha_bar_fn(t1), max_beta))
+    return torch.tensor(betas, dtype=torch.float32)
+
+class ConsistencyDecoderSolver(Solver):
```
Comment (limiteinductive) [inline src/refiners/foundationals/latent_diffusion/solvers/consistency_decoder_solver.py:52]: Is it possible to use instead the existing EulerSolver? If not, what are the differences?

**Review by limiteinductive — COMMENTED** [review]
Thanks for the submission! I have been able to make the conversion works on my side. Could you rebase your branch / remove unecessary files / fix pyright issues. Then two main question for the moment:

 1/ Can we use the existing Euler solver?
 2/ Could you write an e2e test? This will also help me review by having a snippet for inference.

File: src/refiners/foundationals/latent_diffusion/solvers/consistency_decoder_solver.py (hunk 1 above)
Comment (rodSiry) [reply src/refiners/foundationals/latent_diffusion/solvers/consistency_decoder_solver.py:52]: The ConsistencyDecoderVAE comes with a custom, specific scheduler class in diffusers.
I'll double check to see if it boils down to Euler with well chosen inference parameters.

Comment (limiteinductive) [reply src/refiners/foundationals/latent_diffusion/solvers/consistency_decoder_solver.py:52]: Thanks. In any case, you should implement ConsistencyDecoder using Refiners terminology for solvers (we don't use greek letters)

Comment (catwell) [reply src/refiners/foundationals/latent_diffusion/solvers/consistency_decoder_solver.py:53]: :wave: note that we changed the interface of solvers a bit in 0.4 (https://github.com/finegrain-ai/refiners/pull/305) so you will need to rebase and change a few things.

Comment (rodSiry) [reply src/refiners/foundationals/latent_diffusion/solvers/consistency_decoder_solver.py:54]: Hi, 

Sorry for the wait, had a week off, I did the rebase. 

Will re-work the solver and do the e2e tests asap
